molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/impl/AppDeployServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package org.molgenis.app.manager.service.impl;
+
+import org.molgenis.app.manager.exception.AppManagerException;
+import org.molgenis.app.manager.meta.App;
+import org.molgenis.app.manager.meta.AppMetadata;
+import org.molgenis.app.manager.service.AppDeployService;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Query;
+import org.molgenis.data.support.QueryImpl;
+import org.springframework.stereotype.Service;
+import org.springframework.util.FileCopyUtils;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static java.net.URLConnection.guessContentTypeFromName;
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class AppDeployServiceImpl implements AppDeployService
+{
+	private static final String JS_FOLDER = "js";
```

Comment:
Determine mime type based on file extension (just like ordinary spring resources are served from classpath). 
If you pick app serve URI in a way that doesn't clash with the meta methods you can serve all folders.